### PR TITLE
Implemented Safari level effects

### PR DESCRIPTION
**Pull request by RegisCoaxans**
## Description
- We use `(level - 1) / 50` as a modifier which we then adds to each individual escape / catch modifiers. (Thanks Eugene.)
- The chances for an item to drop after catching a pokémon are now increased with Safari levels.
- +1 to eggsteps per step every 10 Safari levels.
Cumulated max bonuses look overkill but remember level 40 is hard to reach, and some pokémons in Friend Safari will need modifiers that high...

## Motivation and Context
Safari zones will be bearable with those changes.

## How Has This Been Tested?
Did some runs and tested that the numbers looked as expected.
We can only test the item changes in the Friend Safari until #4359 is merged.

## Types of changes
- New feature

### CODE REVIEW

**Review by Farboo171 — APPROVED** [review]
Code looks as discussed. Tested and it works between levels 2 and 3 for values changing as anticipated.

File: src/scripts/safari/Safari.ts [hunk 1]
```diff
@@ -347,7 +347,9 @@ class Safari {
     }
 
     public static spawnItemCheck() {
-        if (Rand.boolean()) {
+        const baseChance = 0.4;
+        const levelModifier = (Safari.safariLevel() - 1) / 100;
+        if (Rand.chance(baseChance + levelModifier)) {
```
Comment (Nydaleclya) [inline src/scripts/safari/Safari.ts:352]: Starting with Level 62 this would reduce your chance for an item compared to the previous Level

Comment (RegisCoaxans) [reply src/scripts/safari/Safari.ts:352]: Maximum level is 40, though. I do not think we plan on adding more levels as that one is already long to reach.

**Comment by RegisCoaxans** [comment]
Hopefully made the names less confusing.

**Review by Jaaslet — APPROVED** [review]

**Pull request merged**